### PR TITLE
[17.0][MIG] product_set: Migration to 17.0

### PR DESCRIPTION
**Pull request by NICO-SOLUTIONS**
migration to 17.0

superseeds  #1493

### CODE REVIEW

**Review by CRogos — COMMENTED** [review]
Can you fix and cleanup the commit history?

**Comment by NICO-SOLUTIONS** [comment]
@CRogos 
thanks for the hint. I did not realise that i have pushed 2 MIG-commits.

File: product_set/readme/USAGE.rst
Comment (CRogos) [inline product_set/readme/USAGE.rst:1]: Could you restore the readme files as .md file?

File: product_set/views/product_set.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@
                                 <field name="sequence" widget="handle" />
                                 <field
                                     name="product_id"
-                                    attrs="{'required': [('display_type', '=', False)]}"
+                                    required="display_type == 'False'"
```
Comment (CRogos) [inline product_set/views/product_set.xml]: I think display_type is not "False" but False, or? Therefore I think this should be correct.
`required="not display_type"`

**Review by CRogos — COMMENTED** [review]
Commit history and code looks good now.
Could you readd the readme info as markdown file, and I think there is an error in the xml view.

File: product_set/views/product_set.xml (hunk 1 above)
Comment (NICO-SOLUTIONS) [reply product_set/views/product_set.xml]: yeah right

File: product_set/readme/USAGE.rst
Comment (NICO-SOLUTIONS) [reply product_set/readme/USAGE.rst:1]: not sure what you mean exactly as the files are auto generated.

Comment (CRogos) [reply product_set/readme/USAGE.rst:1]: you're right... not sure what I've seen yesterday. Now it is correct.

**Review by CRogos — APPROVED** [review]

File: product_set/readme/USAGE.rst
Comment (NICO-SOLUTIONS) [reply product_set/readme/USAGE.rst:1]: No worries. Many thanks for your input and the review.

File: product_set/models/product_set.py [hunk 2]
```diff
@@ -32,14 +32,17 @@ class ProductSet(models.Model):
         "it's going to be available for all of them.",
     )
 
-    def name_get(self):
-        return [(rec.id, rec._name_get()) for rec in self]
+    display_name = fields.Char(
+        compute="_compute_display_name", string="Display Name", store=True
```
Comment (pedrobaeza) [inline product_set/models/product_set.py]: I don't think it should be stored, and the name should be translatable.

Comment (NICO-SOLUTIONS) [reply product_set/models/product_set.py]: :thumbsup:

File: product_set/models/product_set.py [hunk 3]
```diff
@@ -32,14 +32,20 @@ class ProductSet(models.Model):
         "it's going to be available for all of them.",
     )
 
-    def name_get(self):
-        return [(rec.id, rec._name_get()) for rec in self]
+    display_name = fields.Char(
+        compute="_compute_display_name",
+        string="Display Name",
+        store=False,
+        translate=True,
```
Comment (pedrobaeza) [inline product_set/models/product_set.py]: These 2 attributes are useless.

Comment (NICO-SOLUTIONS) [reply product_set/models/product_set.py]: perfect... was not sure about the necessity. Now i know it!
Thank you!

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**